### PR TITLE
Add optional ObservedGeneration field to DNSEndpointStatus

### PR DESCRIPTION
**Pull request by shashidharatd**
This is according to the discussion in https://github.com/kubernetes-incubator/external-dns/pull/657#discussion_r208657613

/cc @pmorie @font

### CODE REVIEW

File: pkg/apis/core/v1alpha1/zz_generated.kubebuilder.go [hunk 1]
```diff
@@ -413,7 +413,6 @@ var (
 			Scope: "Namespaced",
 			Validation: &v1beta1.CustomResourceValidation{
 				OpenAPIV3Schema: &v1beta1.JSONSchemaProps{
-					Type: "object",
```
Comment (pmorie) [inline pkg/apis/core/v1alpha1/zz_generated.kubebuilder.go]: Did some changes sneak in unintentionally here? Do we need a new verify task to ensure this doesn't happen again?

**Comment by shashidharatd** [comment]
here is the kubebuilder version which i use, its the same which is in `scripts/download-binaries.sh`
```
root@shashi-laptop:~/gopath/src/github.com/kubernetes-sigs/federation-v2# kubebuilder version
Version: version.Version{KubeBuilderVersion:"1.0.3", KubernetesVendor:"1.11", GitCommit:"3afad1c1fd67ec281c3c6e07508e99d1a3f2aa72", BuildDate:"2018-09-04T19:23:39Z", GoOs:"unknown", GoArch:"unknown"}
```

File: pkg/apis/core/v1alpha1/zz_generated.kubebuilder.go (hunk 1 above)
Comment (shashidharatd) [reply pkg/apis/core/v1alpha1/zz_generated.kubebuilder.go]: This is auto-generated. the code which we have in master probably is generated with older version of kubebuilder.

File: pkg/client/informers/externalversions/factory.go [hunk 2]
```diff
@@ -34,36 +34,79 @@ import (
...
+	customResync     map[reflect.Type]time.Duration
 
 	informers map[reflect.Type]cache.SharedIndexInformer
 	// startedInformers is used for tracking which informers have been started.
 	// This allows Start() to be called multiple times safely.
 	startedInformers map[reflect.Type]bool
 }
 
-// NewSharedInformerFactory constructs a new instance of sharedInformerFactory
+// WithCustomResyncConfig sets a custom resync period for the specified informer types.
```
[the remark this answers is not in the document]
Comment (shashidharatd) [reply pkg/client/informers/externalversions/factory.go]: yes, these are the new generated changes. I am using kubebuilder 1.0.3 as in https://github.com/kubernetes-sigs/federation-v2/blob/master/scripts/download-binaries.sh#L44

File: pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go [hunk 3]
```diff
@@ -51,14 +52,18 @@ type DNSEndpointSpec struct {
 
 // DNSEndpointStatus defines the observed state of DNSEndpoint
 type DNSEndpointStatus struct {
+	// ObservedGeneration is the generation as observed by the controller consuming the DNSEndpoint.
```
[the remark this answers is not in the document]
Comment (shashidharatd) [reply pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go]: @gyliu513, there is nothing more to describe in the comment. the comment is good enough according to me.

**Pull request merged**